validation-test/compiler_crashers_2_fixed/0196-rdar48937223.swift

```diff
@@ -2,10 +2,10 @@
 
 protocol P {}
 
-func fn<T, U: P>(_ arg1: T, arg2: (T) -> U) {} // expected-note {{where 'U' = '()'}}
+func fn<T, U: P>(_ arg1: T, arg2: (T) -> U) {}
 
 func test(str: String) {
-  fn(str) { arg in // expected-error {{global function 'fn(_:arg2:)' requires that '()' conform to 'P'}}
+  fn(str) { arg in // expected-error {{tuple type '()' cannot conform to 'P'; only struct/enum/class types can conform to protocols}}
```

Comment:
In here as well, it would be great to preserve the note which points to declaration.